src/Symfony/Component/HttpKernel/UriSigner.php

```diff
@@ -101,7 +101,7 @@ private function buildUrl(array $url, array $params = []): string
         $pass = isset($url['pass']) ? ':'.$url['pass'] : '';
         $pass = ($user || $pass) ? "$pass@" : '';
         $path = $url['path'] ?? '';
-        $query = isset($url['query']) && $url['query'] ? '?'.$url['query'] : '';
+        $query = $url['query'] ? '?'.$url['query'] : '';
```

Comment:
I think wee can't do that. `query` key is not always present: 
https://www.php.net/manual/en/function.parse-url.php

Reply:
if you look at line 95 above, you will see
```php
$url['query'] = http_build_query($params, '', '&');
```
we set key `query`